gradient_descent.js

```diff
@@ -0,0 +1,53 @@
+console.log("Gradient Descent Algorithm");
+const h = 0.01;
+const x = [1,2];
+const eps = 0.1;
+
+function objective(arr) {
+    ret= 0;
+    for (var i = 0, len = arr.length; i < len; i++) {
+        ret = ret + arr[i] * arr[i];
+    }
+    return ret;
+}
+
+class gradient_descent {
+  constructor(h, objective, x_ini){
+    this.h = h;
+    this.objective = objective;
+    this.x = x_ini;
+    this.delta = 0.1;
+  }
+
+  differentiate(arr){
+    var gradient = [];
+    const x_h = arr.map(el => el);
+    for (var i = 0, len = arr.length; i < len; i++) {
+        x_h[i] = arr[i] + this.h;
+        gradient.push((this.objective(x_h) - this.objective(arr))/this.h);
+        x_h[i] = arr[i];
+    }
+    return gradient;
+  }
```

Comment:
You can use map here:
```suggestion
differentiate(arr) {
    const x_h = arr.map(el => el);
    return arr.map((el, i) => {
        x_h[i] = el + this.h;
        gradient = (this.objective(x_h) - this.objective(arr))/this.h);
        x_h[i] = el;
        return gradient;
    })
}
```